builder/tests/test_actions.py

```diff
@@ -90,60 +90,116 @@ def test_duplicate_search(version_from_scratch):
+    )
+    s2 = actions.create_search(
+        draft=draft, term="Soft tissue lesion", codes={"239964003"}
+    )
+
+    # Act: delete the search for "Soft tissue lesion"
+    actions.delete_search(search=s2)
+
+    # Assert...
+    # that the codelist has only 1 search
+    assert draft.searches.count() == 1
+    # that the codelist has 2 codes
+    assert draft.code_objs.count() == 2
+
+    # Arrange: recreate the search for "Soft tissue lesion"
+    actions.create_search(draft=draft, term="Soft tissue lesion", codes=["239964003"])
+
+    # Act: delete the search for "epicondylitis"
+    actions.delete_search(search=s1)
+
+    # Assert...
+    # that the codelist has only 1 search
+    assert draft.searches.count() == 1
+    # that the codelist now has only 1 code
+    assert draft.code_objs.count() == 1
+
+
+def test_delete_search_codelist_with_codes(version_with_no_searches):
+    draft = version_with_no_searches
+    starting_code_count = draft.code_objs.count()
+    assert starting_code_count > 0
+
+    # This codelist has all the codes for disorder of elbow excluding arthritis
+    # Epicondylitis codes are already on the codelist
+    epicondylitis_codes = {"73583000", "202855006"}
+    # Tennis toe code is not on the codelist
+    tennis_toe_codes = {"238484001"}
+    starting_codes = draft.code_objs.values_list("code", flat=True)
+    for code in epicondylitis_codes:
+        assert code in starting_codes
+    for code in tennis_toe_codes:
+        assert code not in starting_codes
+
+    # Act: create a term search
+    s = actions.create_search(
+        draft=draft,
+        term="Tennis tow",
+        codes=tennis_toe_codes,
+    )
+
+    # Act: delete the search
+    actions.delete_search(search=s)
+
+    # Assert...
+    # that the codelist has 0 searches
+    assert draft.searches.count() == 0
+    # that the codelist still has the starting codes that were not associated with the search
+    assert draft.code_objs.count() == starting_code_count
+
+    # Arrange: create new searches
+    s1 = actions.create_search(
+        draft=draft,
+        term="epicondylitis",
+        codes=epicondylitis_codes,
+    )
+    s2 = actions.create_search(draft=draft, term="Tennis toe", codes=tennis_toe_codes)
+
+    # Act: delete the search for "Tennis toe"
+    actions.delete_search(search=s2)
+
+    # Assert...
+    # that the codelist has only 1 search
+    assert draft.searches.count() == 1
+    # that the codelist still has the starting codes that were not associated with the search
+    assert draft.code_objs.count() == starting_code_count
+
+    # Act: delete the search for "epicondylitis"
+    actions.delete_search(search=s1)
+
+    # Assert...
+    # that the codelist has no searches
+    assert draft.searches.count() == 0
+    # This search matched 2 starting codes which are now deleted.  Others codes that were not
+    # associated with the search remain
+    assert draft.code_objs.count() == starting_code_count - 2
```

Comment:
Is this what should happen?  The 2 epicondylitis codes were on the codelist when it had no searches, the search means that they become associated with the search, so when you delete it, those 2 codes get deleted.

Should the delete action check the status of the `code_obj` and only delete it if it's unresolved?